src/features/chat/index.ts

```diff
@@ -0,0 +1,36 @@
+import { NotificationHandler, RequestHandler } from 'vscode-languageserver'
+import {
+    ChatParams,
+    ChatResult,
+    EndChatParams,
+    EndChatResult,
+    QuickActionParams,
+    TabEventParams,
+    VoteParams,
+} from './types'
+
+/**
+ * The Chat feature interface. Provides access to chat features
+ */
+export type Chat = {
+    onChatPrompt: (handler: RequestHandler<ChatParams, ChatResult | undefined | null, void>) => void // send result as partials and then send complete message
+    onEndChat: (handler: RequestHandler<EndChatParams, EndChatResult, void>) => void
+    onQuickAction: (handler: RequestHandler<QuickActionParams, ChatResult, void>) => void
+
+    // Notifications
+    onSendFeedback?: (handler: NotificationHandler<VoteParams>) => void
+    onReady?: (handler: NotificationHandler<void>) => void
+    onTabAdd?: (handler: NotificationHandler<TabEventParams>) => void
+    onTabChange?: (handler: NotificationHandler<TabEventParams>) => void
+    onTabRemove?: (handler: NotificationHandler<TabEventParams>) => void
+    onVote?: (handler: NotificationHandler<VoteParams>) => void
```

Comment:
How would `onVote` be different from `onSendFeedback` given they use the same parameters type?

Reply:
They share the same input shape but correspond to different events in the UI. I will create an explicit interface for feedback params that will be same as voteparams for the time being. This should allow future flexibility and explicit naming

Reply:
Actually it was a mistake on my part, they have different shapes I will update it